week3_HM/todo/src/components/TodoList.js

```diff
@@ -0,0 +1,79 @@
+import React, { Component } from 'react'
+import input from './buttons'
+import InsertTasks from './insertTasks'
+//import DatePicker from 'react-datepicker';
+//import moment from 'moment';
+
+const data = require('../todo.json')
+
+export default class TodoList extends Component{
+    state = {
+     mineTasks: data
+    }
+    handleChecked = (index) => {
+        const mineTasks = this.state.mineTasks
+        mineTasks[index].done = !mineTasks[index].done
+        this.setState({
+            mineTasks
+       })
+    }
+    handleAddingNewTask = (text, date) => {
+        console.log(text, date)
+        const ids = this.state.mineTasks.map(task => task.id)
+        const newId = Math.max(...ids) + 1
+        const newTask = {
+            "id": newId,
+            "description": text,
+            "deadline": date,
+            "done": false
+        }
+        this.setState({
+            mineTasks: [
+                ...this.state.mineTasks,
+                newTask]
+        })
+    }
+    handleDeleteTask = (index) => {
+        console.log("deleting")
+        const mineTasks = this.state.mineTasks
+        const newMIneTasks = mineTasks.splice(index  , 1)
+        this.setState({
+        newMIneTasks
+    })
```

Comment:
Another indentation issue. Take a look at eslint, it will handle these sorts of problems for you by producing warnings.